flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/TableResult.java

```diff
@@ -131,14 +131,27 @@
 	 *      it... // collect same data
 	 *  }
 	 * }</pre>
+	 *
+	 * <p>This method guarantees end-to-end exactly-once record delivery
+	 * which requires the checkpointing mechanism to be enabled.
+	 * By default, checkpointing is disabled. To enable checkpointing,
+	 * call `StreamExecutionEnvironment#enableCheckpointing()` method.
+	 *
+	 * <p>Only this method or {@link #print()} method can be called for a TableResult instance,
+	 * because the result can only be accessed once.
```

Comment:
NOTE: In order to fetch result, you can call either {@link #print()} or {@link #collect()}. But, they can't be called both on the same {@link TableResult} instance, because the result can only be accessed once.


What do you think?